### PR TITLE
Test to confirm empty values are applied to select and multiselect fields

### LINKED ISSUE #10992: Contacts API "Select - Multiple" field - set empty value
I am trying to set an empty value for a "Select - Multiple" field and it does not work. It works for not-empty values, but it does not work with an empty array or null (which should be the logical candidates for an empty value).

Here is the problem in details:

1. I created a field in mautic with the type "Select - Multiple" and added 2 possible vales.
2. I am updating the values with a PHP script using mautic/api-library of course.

The problem is that when I am updating the field values of a specific contact with actual values - it all works, but when I post an empty array or NULL - the value does not change.

this works:
```php
    $data = [
      'multivalue_categories' => ['cat1'],
    ];

      $mauticUser = $contactApi->edit(
        MAUTIC_ID,
        $data,
        FALSE
      );
```

this does not work:
```php
    $data = [
      'multivalue_categories' => [],
    ];

      $mauticUser = $contactApi->edit(
        MAUTIC_ID,
        $data,
        FALSE
      );
```

Then I changed the value in Mautic (of the contact) (meaning - removed all categories ) and fetched the user and I saw that the value then is NULL - then I tried also this:
```php
      $data = [
        'multivalue_categories' => null,
      ];

      $mauticUser = $contactApi->edit(
        MAUTIC_ID,
        $data,
        FALSE
      );
```
but that did not work either - it does not produce an error, but it does not change it either.
On the other hand I tried also "" or null as a string - but that already produced an error
`Response: {"errors":[{"code":400,`

Can someone please help me how to set empty value to a "Select - Multiple" field, because all the logical options are not working...

<bountysource-plugin>

---
Want to back this issue? **[Post a bounty on it!](https://app.bountysource.com/issues/108499009-contacts-api-select-multiple-field-set-empty-value?utm_campaign=plugin&utm_content=tracker%2F5355074&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://app.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F5355074&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by biozshock**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | ✔️ 
| New feature/enhancement? (use the a.x branch)      | ✔️
| Deprecations?                          | ❌
| BC breaks? (use the c.x branch)        | ❌
| Automated tests included?              | ✔️ 
| Issue(s) addressed                     | Fixes #10992  

## Description
Add a test that checks the empty values for custom select and multiselect fields are applied.

---
### 📋 Steps to test this PR:

1. Tests are ok.

### CODE REVIEW

**Comment by biozshock** [comment]
@escopecz  That's the last one i do want to get into the M6.

**Review by escopecz — APPROVED** [review]
Perfect! Merging right-away as there is no production file change. Just added test 👍

File: app/bundles/LeadBundle/Tests/Controller/Api/FieldApiControllerFunctionalTest.php [hunk 1]
```diff
@@ -170,6 +174,232 @@ public function dataForCreatingNewBooleanFieldApiEndpoint(): iterable
...
+                        'label' => 'Halusky',
+                        'value' => 'halusky',
+                    ],
+                    [
+                        'label' => 'Bramborak',
+                        'value' => 'bramborak',
+                    ],
+                    [
+                        'label' => 'Makovec',
+                        'value' => 'makovec',
```
Comment (escopecz) [inline app/bundles/LeadBundle/Tests/Controller/Api/FieldApiControllerFunctionalTest.php:208]: Getting hungry while reading it 😆
